### PR TITLE
docs: destructure `req` in middleware

### PR DESCRIPTION
**Pull request by danielroe**
Not super familiar with SolidStart but as-is this gives a type error.

### CODE REVIEW

**Comment by juliusmarminge** [comment]
i thought destructuring will actually consume the signal? idk, signals are weird and i'll never see how they provide a better dx 🤷🏼

**Comment by danielroe** [comment]
Then should it be `req.req`?

**Comment by juliusmarminge** [comment]
> Then should it be `req.req`?

Oh right. Yes.

`req` here must be old. It's an object now with the request inside of it (since 5.0.0). I think we refer to it as `opts` in the rest

```ts
.middleware(opts => opts.req)
```

File: docs/src/pages/solid.mdx [hunk 1]
```diff
@@ -88,7 +88,7 @@ export const uploadRouter = {
       maxFileSize: "1MB",
     },
   })
-    .middleware((req) => {
+    .middleware(({ req }) => {
       const h = req.headers.get("someProperty");
```
Comment (juliusmarminge) [inline docs/src/pages/solid.mdx:92]:
```suggestion
    .middleware((opts) => {
      const h = opts.req.headers.get("someProperty");
```

**Review by juliusmarminge — APPROVED** [review]

**Pull request merged**